cpp/email_watcher.cc

```diff
@@ -0,0 +1,258 @@
+                                  subject, message_body, /* priority = */EmailSender::VERY_HIGH, EmailSender::PLAIN_TEXT,
+                                  /* reply_to = */"", /* use_authentication = */true, /* use_ssl = */true, attachments);
+}
+
+
+const std::string MAPFILE_PATH(UBTools::GetTuelibPath() + "email_watcher_last_seen_times.map");
+
+
+std::map<std::string, time_t> LoadSectionNamesToLastSeenTimeWindowsMap() {
+    std::map<std::string, time_t> section_names_to_last_seen_time_map;
+    if (FileUtil::Exists(MAPFILE_PATH)) {
+        const auto input(FileUtil::OpenInputFileOrDie(MAPFILE_PATH));
+        unsigned current_line_number(0);
+        while (not input->eof()) {
+            ++current_line_number;
+            const auto line(input->getline());
+            const auto first_equal_pos(line.find('='));
+            if (unlikely(first_equal_pos == std::string::npos or first_equal_pos == 0))
+                LOG_ERROR("missing or leading equal-sign in " + MAPFILE_PATH + " on line #"
+                          + std::to_string(current_line_number) + "!");
+            unsigned last_seen_time;
+            if (unlikely(not StringUtil::ToNumber(line.substr(first_equal_pos + 1), &last_seen_time)))
+                LOG_ERROR("can't convert \"" + line.substr(first_equal_pos + 1) + "\" to an unsigned number! (line #"
+                          + std::to_string(current_line_number) + " in " + MAPFILE_PATH + ")");
+            section_names_to_last_seen_time_map[line.substr(0, first_equal_pos)] = static_cast<time_t>(last_seen_time);
+        }
+    }
+
+    LOG_INFO("Loaded " + std::to_string(section_names_to_last_seen_time_map.size()) + " mappings from " + MAPFILE_PATH + ".");
+
+    return section_names_to_last_seen_time_map;
+}
+
+
+void SaveSectionNamesToLastSeenTimeWindowsMap(const std::map<std::string, time_t> &section_names_to_last_seen_time_map) {
+    const auto output(FileUtil::OpenOutputFileOrDie(MAPFILE_PATH));
+    for (const auto &[section_name, last_seen_time] : section_names_to_last_seen_time_map)
+        (*output) << section_name << '=' << last_seen_time << '\n';
+    LOG_INFO("Wrote " + std::to_string(section_names_to_last_seen_time_map.size()) + " entries/entry to " + MAPFILE_PATH + ".");
+}
+
+
+} // unnamed namespace
+
+
+int Main(int argc, char *argv[]) {
+    if (argc != 3)
+        ::Usage("ini_file_path mbox_path");
+
+    const IniFile ini_file(argv[1]);
+
+    std::vector<std::string> notification_email_addresses;
+    if (StringUtil::SplitThenTrimWhite(ini_file.getString("", "notify"), ',', &notification_email_addresses) == 0)
+        LOG_ERROR("we need at least one notification email address in the \"notify\" entry in \""
+                  + ini_file.getFilename() + "\"!");
+
+    const std::string backup_dir_path(ini_file.getString("", "backup_dir_path") + "/");
+
+    const auto overdue_db(OpenOrCreateOverdueDatabase());
```

Comment:
Is this connection actually used anywhere?

Reply:
Apparently not.

Reply:
Seems you removed it now.

In which interval will this program be called? Is it necessary to prevent it from sending multiple overdue reminders for the same mail?

Reply:
It will probably run every 10 minutes or so.  And no, because we update the last sent time when we complain.